src/Apps/ArtAdvisor/07-Curated-Discovery/weaviate-db.ts

```diff
@@ -60,6 +66,51 @@ export class WeaviateDB {
 
     return result as DiscoveryMarketingCollections[]
   }
+
+  /**
+   * Fetch articles by semantic search on an array of concepts.
+   */
+  async getNearArticles({
+    concepts,
+    limit = 10,
+  }: {
+    /** List of concepts for semantic search */
+    concepts: string[]
+    /** Max number of articles to return */
+    limit?: number
+  }) {
+    console.log("[WeaviateDB] getNearArticles", {
+      concepts,
+      limit,
+    })
+
+    const conceptArray = ensureValidConcepts(concepts)
+
+    const response = await this.client.graphql
+      .get()
+      .withClassName(this.articlesClass)
+      .withNearText({
+        concepts: conceptArray,
+      })
+      .withLimit(limit)
+      .withFields(
+        "internalID title articleDescription href imageUrl _additional { id distance }"
+      )
+      .do()
+
+    const result: ArticleType[] = response.data.Get[DEFAULT_ARTICLES_CLASS]
```

Comment:
If we're setting the articles class in the config, then technically this would be more correct:

```suggestion
    const result: ArticleType[] = response.data.Get[this.articlesClass]
```

(Same idea on L65 above, my bad for not noticing in the earlier PR)

Reply:
Ah, nice catch. I will revert back to what we had before. This was just a quick hack while debugging.

Reply:
That was a confusing response by me. But have changed both to your suggestion in [430d71b](https://github.com/artsy/force/pull/14064/commits/430d71b846575e3f1c458cb5b842f532f84e1866).